packages/lodestar/src/api/impl/validator/produceAttestation.ts

```diff
@@ -12,11 +12,12 @@ export async function produceAttestation(
   slot: Slot
 ): Promise<Attestation|null> {
   try {
-    const [headState, headBlock, validatorIndex] = await Promise.all([
-      clone(chain.latestState, config.types.BeaconState) as BeaconState,
+    const [headBlock, validatorIndex] = await Promise.all([
       db.block.get(chain.forkChoice.head()),
       db.getValidatorIndex(validatorPubKey)
     ]);
+    const headState = await db.state.get(headBlock.stateRoot);
+    await processSlots(config, headState, slot);
```

Comment:
I don't think we need this `processSlots` here?
It could go in `assembleAttestation` instead of the while loop at the top.

Reply:
Got that `processSlots` in https://github.com/ethereum/eth2.0-specs/blob/dev/specs/validator/0_beacon-chain-validator.md
```
First, the validator should construct attestation_data, an AttestationData object based upon the state at the assigned slot.

Let head_block be the result of running the fork choice during the assigned slot.
Let head_state be the state of head_block processed through any empty slots up to the assigned slot using process_slots(state, slot).
```

Reply:
Ok, we can remove the loop at the top of `assembleAttestation` then, since we're doing the same here.